src/components/Banner.vue

```diff
@@ -0,0 +1,208 @@
+<script setup>
+import { ref, computed } from "vue";
+import useMovieStore from "../store/movieStore";
+
+const movieStore = useMovieStore();
+
+const searchByTitle = ref(true);
+const query = ref('');
+const selectedMovie = computed(() => movieStore.getMovieById);
+
+const searchHandler = (option) => {
+  if (option === "title") searchByTitle.value = true;
+  else if (option === "genres") searchByTitle.value = false;
+  else searchByTitle.value = true;
+  movieStore.setFilterParam(option);
+};
+
+const clickHandler = () => {
+  movieStore.setSearchQuery(query.value);
+}
+
+</script>
+
+<template>
+  <div class="root">
+    <header>
+      <div class="header__left">
+        <span class="header__text1">netflix</span>
+        <span>roulette</span>
+      </div>
+      <div v-if="movieStore.isMovieDetailOpen" class="header__right">
+        <img src="../assets/search.svg" @click="movieStore.toggleMovieDetailPanel(null)">
+      </div>
+    </header>
+    <div v-if="movieStore.isMovieDetailOpen" class="banner">
+      <img class="banner-img"
+        :src="selectedMovie.posterurl"
+        :placeholder="selectedMovie.title" />
+      <div class="banner-info">
+        <div>
+          <span>{{ selectedMovie.title }}</span>
+          <span class="banner-info__rating">{{ selectedMovie.imdbRating }}</span>
+        </div>
+        <div class="banner-info__year">
+          <span>{{ selectedMovie.year }}</span>
+          <span>{{ selectedMovie.duration.slice(2).replace('M', '') }} min</span>
+        </div>
+        <div class="banner-info__storyline">{{ selectedMovie.storyline }}</div>
+      </div>
+    </div>
+    <div v-else class="panel">
+      <div class="searchlabel">FIND YOUR MOVIE</div>
+      <div class="searchbar">
```

Comment:
Use form for parent, so you can submit by pressing an "Enter" button while focusing on the input field:

```html
      <form class="searchbar" @submit.prevent="clickHandler()">
        <input v-model="query" placeholder="Type and Search" />
        <button type="submit">SEARCH</button>
      </form>
```